### PR TITLE
Improve test coverage for is_typeddict

### PR DESCRIPTION
**Pull request by JelleZijlstra**
In particular, it's important to test that is_typeddict(TypedDict)
returns False.

### CODE REVIEW

File: Lib/test/test_typing.py [hunk 1]
```diff
@@ -7223,10 +7223,29 @@ class Wrong(*bases):
                         pass
 
     def test_is_typeddict(self):
-        assert is_typeddict(Point2D) is True
-        assert is_typeddict(Union[str, int]) is False
+        self.assertTrue(is_typeddict(Point2D))
+        self.assertFalse(is_typeddict(Union[str, int]))
```
Comment (AlexWaygood) [inline Lib/test/test_typing.py]: +1 for using the unittest methods, but note that this isn't a _direct_ translation. `assertTrue` only asserts that the thing is truthy, not that it actually is the `True` constant.

**Review by AlexWaygood — APPROVED** [review]

File: Lib/test/test_typing.py (hunk 1 above)
Comment (JelleZijlstra) [reply Lib/test/test_typing.py]: Oh good point, I'll switch to `assertIs(..., True)`.

File: Lib/test/test_typing.py [hunk 2]
```diff
@@ -7223,10 +7223,29 @@ class Wrong(*bases):
...
+        self.assertIs(is_typeddict(call_based()), False)
+
+        T = TypeVar("T")
+        class BarGeneric(TypedDict, Generic[T]):
+            a: T
+        self.assertIs(is_typeddict(BarGeneric), True)
+        self.assertIs(is_typeddict(BarGeneric[int]), False)
+        self.assertIs(is_typeddict(BarGeneric()), False)
+
+        class NewGeneric[T](TypedDict):
```
Comment (JelleZijlstra) [inline Lib/test/test_typing.py:7241]: I'll manually remove this one from the 3.11 backport

**Comment by AlexWaygood** [comment]
We should probably backport these to `typing_extensions` too

**Comment by JelleZijlstra** [comment]
Maybe we should get Miss Islington to do that too :)

**Pull request merged**